Server/Internal/Server/Api.cs

```diff
@@ -0,0 +1,16 @@
+using Grpc.Core;
+using Greet;
+
+namespace Greet.Services;
+
+public partial class GreeterService : MyGreeter.MyGreeterBase
+{
+    private readonly ILogger<GreeterService> _logger;
+    private readonly ServerOptions _options;
+    
+    public GreeterService(ILogger<GreeterService> logger, ServerOptions options)

```

Comment:
Why do we need to pass in a logger as a parameter? In the current version, we create a new logger and pass it to client/server interceptors. From there, we can pull out the logger from ctx to use in our code stubs.